frontend/src/pages/Friends.jsx

```diff
@@ -1,17 +1,216 @@
 import React, { Component } from 'react';
+import {connect} from 'react-redux';
+import * as FriendsActions from "../actions/FriendsActions";
 import 'semantic-ui-css/semantic.min.css';
+import FriendListComponent from '../components/FriendsListComponent';
+import "./styles/Friends.css";
+import { Button } from 'semantic-ui-react';
+import store from "../store/index";
+import HTTPFetchUtil from "../util/HTTPFetchUtil";
+import { SemanticToastContainer, toast } from 'react-semantic-toasts';
+import 'react-semantic-toasts/styles/react-semantic-alert.css';
+import Cookies from 'js-cookie';
+import utils from "../util/utils";
 
-class Friends extends Component {	
+class Friends extends Component {
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			userIdFullURL: null,
+			isLoggedIn: false,
+			userId: null,
+			listData: null,
+			mode: "friends",
+			hostName: "",
+			friendButtonColor: "teal",
+			requestButtonColor: "grey",
+		}
+		this.removeFriend = this.removeFriend.bind(this);
+		this.approveFriendRequest = this.approveFriendRequest.bind(this);
+	}
+	
+	componentDidMount(){
+		let userIdShortString = utils.getShortAuthorId(Cookies.get("userID")) || store.getState().loginReducers.authorId;
```

Comment:
use store.getState().loginReducers.userId, remove authorId from the store since it is a duplication